### PR TITLE
CB-23-Possibility-to-remove-recipe

### PR DESCRIPTION
**Pull request by banaskrzysztof**
https://team-1626248039387.atlassian.net/browse/CB-23?atlOrigin=eyJpIjoiYzNiMmFlMjNmZGExNDBjMTgzNTFlZWYzNGMxY2EyMDgiLCJwIjoiaiJ9

### CODE REVIEW

File: src/app/modules/recipe-detail/components/recipe-detail-dialog.component.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+@Component({
+  selector: 'app-recipe-detail-dialog',
+  templateUrl: './recipe-detail-dialog.component.html',
+  styleUrls: ['./recipe-detail-dialog.component.scss']
+})
+
+export class RecipeDetailDialogComponent {
+
+  constructor() {}
```
Comment (tomekknowde) [inline src/app/modules/recipe-detail/components/recipe-detail-dialog.component.ts:11]: remove empty `constructor`

File: src/app/modules/recipe-detail/containers/recipe-detail.component.html [hunk 2]
```diff
@@ -28,7 +28,7 @@
   <div class="box__actions">
    <span> <mat-icon aria-hidden="true">chat_bubble</mat-icon>
     5 komentarzy</span>
-    <span><mat-icon aria-hidden="true">delete</mat-icon>
+    <span id="delete" (click)="openDialog(recipe.id)"><mat-icon aria-hidden="true">delete</mat-icon>
```
Comment (tomekknowde) [inline src/app/modules/recipe-detail/containers/recipe-detail.component.html:31]: It should be a class `delete-btn` or `btn btn-delete` or `button-delete` or `delete-button` whatever, but not an id 😄

File: src/app/modules/recipe-detail/containers/recipe-detail.component.scss [hunk 3]
```diff
@@ -87,3 +87,7 @@ a {
     padding-bottom: 30px;
   }
 }
+
+#delete {
```
Comment (tomekknowde) [inline src/app/modules/recipe-detail/containers/recipe-detail.component.scss:91]: use class

File: src/app/modules/recipe-detail/containers/recipe-detail.component.ts [hunk 4]
```diff
@@ -14,14 +16,24 @@ export class RecipeDetailComponent implements OnInit {
...
+              public dialog: MatDialog) {
   }
 
   ngOnInit(): void {
     this.activeRoute.params.subscribe(params => {
       this.getRecipe(params['id']);
     });
+  }
+
+  openDialog(id: number) {
```
Comment (tomekknowde) [inline src/app/modules/recipe-detail/containers/recipe-detail.component.ts:29]: return type

File: src/app/modules/recipe-detail/containers/recipe-detail.component.ts [hunk 5]
```diff
@@ -30,4 +42,8 @@ export class RecipeDetailComponent implements OnInit {
     })
   }
 
+  public deleteRecipe(id: number): void {
+    this.recipeDetailService.deleteRecipe(id).subscribe(recipe => {
+    });
```
Comment (tomekknowde) [inline src/app/modules/recipe-detail/containers/recipe-detail.component.ts:47]:
```suggestion
    this.recipeDetailService.deleteRecipe(id).subscribe();
```

**Review by KrzysiekLipiec — APPROVED** [review]

**Review by tomekknowde — APPROVED** [review]

**Pull request merged**